### PR TITLE
add foreman_ prefix to role variables

### LINKED ISSUE #1118: Role Variable Prefix
##### SUMMARY
This issue is to open the discussion around variable naming convention for roles. Currently, "generic" variable names are used (eg. `server_url`, `username`, `manifest`, etc). This could lead to conflicting variables during runtime between other roles, facts, or extra_vars. I propose roles use variables with the prefix `foreman_` to "namespace" vars used by these roles.

Current state example:
```
server_url: https://satellite.example.com
username: admin
password: password
organization: "Default Organization"
manifest_path: /tmp/manifest.zip
activation_keys:
          - name: "Basic Activation Key"
```

Future State Example:
```
foreman_server_url: https://satellite.example.com
foreman_username: admin
foreman_password: password
foreman_organization: "Default Organization"

foreman_manifest_path: /tmp/manifest.zip

foreman_activation_keys:
          - name: "Basic Activation Key"
```

##### ISSUE TYPE
 - Feature Idea

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: #1118

### CODE REVIEW

**Comment by evgeni** [comment]
@ehelms @mdellweg @wbclark I need y'alls opinion on one thing here:

This is technically a breaking change, so I guess we should bump the version to 2.0.0-dev in this PR?

File: roles/manifest/README.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ Role Variables
...
-- `manifest_download`: Whether to first download the Manifest from the Red Hat Customer Portal. Defaults to `False`.
-- `manifest_uuid`: UUID of the Manifest to download, corresponding to a [Subscription Allocation](https://access.redhat.com/management/subscription_allocations) defined on your Red Hat account. Required when `manifest_download` is `True`.
+- `foreman_manifest_path`: Path to subscription Manifest file on Ansible target host. When using `manifest_download`, it is first downloaded to this location from the Red Hat Customer Portal before being uploaded to the Foreman server.
+- `foreman_manifest_download`: Whether to first download the Manifest from the Red Hat Customer Portal. Defaults to `False`.
+- `foreman_manifest_uuid`: UUID of the Manifest to download, corresponding to a [Subscription Allocation](https://access.redhat.com/management/subscription_allocations) defined on your Red Hat account. Required when `manifest_download` is `True`.
 - `rhsm_username`: Your username for the Red Hat Customer Portal. Required when `manifest_download` is `True`.
 - `rhsm_password`: Your password for the Red Hat Customer Portal. Required when `manifest_download` is `True`.
```
Comment (evgeni) [inline roles/manifest/README.md]: I didn't rename those two, but I guess we should.

`foreman_rhsm_username` just sounds so wrong…

**Comment by evgeni** [comment]
and this obviously lacking a changelog snippet

File: roles/manifest/README.md [hunk 2]
```diff
@@ -8,9 +8,9 @@ Role Variables
...
 
-- `manifest_path`: Path to subscription Manifest file on Ansible target host. When using `manifest_download`, it is first downloaded to this location from the Red Hat Customer Portal before being uploaded to the Foreman server.
-- `manifest_download`: Whether to first download the Manifest from the Red Hat Customer Portal. Defaults to `False`.
-- `manifest_uuid`: UUID of the Manifest to download, corresponding to a [Subscription Allocation](https://access.redhat.com/management/subscription_allocations) defined on your Red Hat account. Required when `manifest_download` is `True`.
+- `foreman_manifest_path`: Path to subscription Manifest file on Ansible target host. When using `manifest_download`, it is first downloaded to this location from the Red Hat Customer Portal before being uploaded to the Foreman server.
+- `foreman_manifest_download`: Whether to first download the Manifest from the Red Hat Customer Portal. Defaults to `False`.
+- `foreman_manifest_uuid`: UUID of the Manifest to download, corresponding to a [Subscription Allocation](https://access.redhat.com/management/subscription_allocations) defined on your Red Hat account. Required when `manifest_download` is `True`.
```
Comment (mdellweg) [inline roles/manifest/README.md:13]:
```suggestion
- `foreman_manifest_path`: Path to subscription Manifest file on Ansible target host. When using `foreman_manifest_download`, it is first downloaded to this location from the Red Hat Customer Portal before being uploaded to the Foreman server.
- `foreman_manifest_download`: Whether to first download the Manifest from the Red Hat Customer Portal. Defaults to `False`.
- `foreman_manifest_uuid`: UUID of the Manifest to download, corresponding to a [Subscription Allocation](https://access.redhat.com/management/subscription_allocations) defined on your Red Hat account. Required when `foreman_manifest_download` is `True`.
```

File: roles/manifest/README.md (hunk 1 above)
Comment (mdellweg) [reply roles/manifest/README.md]: We definitely should. `whatever_username` and `whatever_password` are the primary examples for collisions when we don't namespace properly.

Comment (mdellweg) [reply roles/manifest/README.md]:
```suggestion
- `rhsm_username`: Your username for the Red Hat Customer Portal. Required when `manifest_download` is `True`.
- `rhsm_password`: Your password for the Red Hat Customer Portal. Required when `manifest_download` is `foreman_true`.
```

Just kidding. ;) But should "true" not be lowercase in yaml?

Comment (evgeni) [reply roles/manifest/README.md]: should, yes, `ansible-lint` will yell at you, but it accepts both, can update when I touch this, sure

**Comment by ehelms** [comment]
Aye, update to 2.0 for this change. And a changelog entry for this would be good.

**Comment by mdellweg** [comment]
Oh yeah, 2.0. Forgot to answer about that one.

**Comment by evgeni** [comment]
@mdellweg @ehelms updated :)

**Review by mdellweg — APPROVED** [review]
I see more True False, but that does not need to block us here.

**Pull request merged**